plugins/aws/cache_utils.go

```diff
@@ -0,0 +1,53 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+const (
+	mfaCacheKeyID        = "sts-mfa"
+	assumeRoleCacheKeyID = "sts-assume-role"
+)
+
+// stsCacheWriter writes aws temp credentials to cache using the awsCacheKey
+type stsCacheWriter struct {
+	awsCacheKey string
+	cache       sdk.CacheOperations
+}
+
+func (c stsCacheWriter) persist(credentials aws.Credentials) error {
+	return c.cache.Put(c.awsCacheKey, credentials, credentials.Expires)
+}
+
+func NewStsCacheWriter(key string, cache sdk.CacheOperations) stsCacheWriter {
+	return stsCacheWriter{
+		awsCacheKey: key,
+		cache:       cache,
+	}
+}
+
+type stsCacheProvider struct {
+	awsCacheKey string
+	cache       sdk.CacheState
+}
+
+func (c stsCacheProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+	var cached aws.Credentials
+	if ok := c.cache.Get(c.awsCacheKey, &cached); ok {
+		return cached, nil
+	}
+
+	return aws.Credentials{}, fmt.Errorf("did not find cached credentials")
+}
+
+func getRoleCacheKey(roleArn string, accessKeyID string) string {
+	return assumeRoleCacheKeyID + accessKeyID + roleArn
+}
+
+func getMfaCacheKey(accessKeyID string) string {
+	return mfaCacheKeyID + accessKeyID
```

Comment:
nit: it's generally a good practice to use a separatator between these elements. In this case it is not likely to lead to any problems because of the specific format of the Access Key ID and Role ARN, but at least it makes the cache keys easier to interpret.

For example:
```
fmt.Sprintf("%s|%s|%s", assumeRoleCacheKeyID, accessKeyID, roleArn)
```

Reply:
Done!